### PR TITLE
Keep track and report the unit files that contain test symbols

### PR DESCRIPTION
**Pull request by akyrtzi**
This allows the client to be much more efficient for handling unit test symbol changes.

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: lib/Database/ImportTransactionImpl.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ class ImportTransaction::Implementation {
 
   IDCode getUnitCode(StringRef unitName);
   IDCode addProviderName(StringRef name, bool *wasInserted);
+  // Marks a provider as containing test symbols.
+  void markProviderWithTestSymbols(IDCode provider);
+  bool isProviderContainingTestSymbols(IDCode provider);
```
Comment (benlangmuir) [inline lib/Database/ImportTransactionImpl.h]: I suggest the following names:
```
setProviderContainsTestSymbols
providerContainsTestSymbols
```

Comment (akyrtzi) [reply lib/Database/ImportTransactionImpl.h]: Done.

**Pull request merged**